org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
>I still don't see any issues with my [] at the end detection

Note: current problem (mix of `int4[]` vs `_int4` in master) comes from the fact that pgjdbc is trying to cache `oid -> typename` information here: https://github.com/pgjdbc/pgjdbc/blob/7311b4ef160231780323573911c1543d0515340b/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L375

1) One wants to resolve "what is the oid for `int4[]`"?
2) It queries the database, and comes up with oid 1007
3) pgjdbc caches that 1007 maps to `int4[]` <-- this is what corrupts typename cache contents

>although I still don't see any issues with my [] at the end detection

Well, `[]` is good for `isArray` detection. My original question was "are there any other reasons to use `[]` other than `isArray`"? I'm not sure what is the proper way to reframe "typenamecache vs array" logic, and I might consider taking a step back to avoid possibility of `[]` sneaking into pgjdbc outputs.

> dropping the cache seems fine (not sure what are the perf implications of that).

That is why I consider to reverting to some underscore black magic.